lib/libc/string/strspn.c

```diff
@@ -39,34 +39,24 @@ __FBSDID("$FreeBSD$");
 size_t
 strspn(const char *s, const char *charset)
 {
-	/*
-	 * NB: idx and bit are temporaries whose use causes gcc 3.4.2 to
-	 * generate better code.  Without them, gcc gets a little confused.
-	 */
 	const char *s1;
-	u_long bit;
 	u_long tbl[(UCHAR_MAX + 1) / LONG_BIT];
-	int idx;
 
 	if(*s == '\0')
 		return (0);
 
 #if LONG_BIT == 64	/* always better to unroll on 64-bit architectures */
 	tbl[3] = tbl[2] = tbl[1] = tbl[0] = 0;
 #else
-	for (idx = 0; idx < sizeof(tbl) / sizeof(tbl[0]); idx++)
+	for (unsigned int idx = 0; idx < sizeof(tbl) / sizeof(tbl[0]); idx++)
```

Comment:
size_t, and I'm unconvinced we need the manual unrolling, every compiler of interest on godbolt will unroll this loop, it's such a basic optimisation to make here.